internal/storage/location.go

```diff
@@ -12,92 +12,47 @@ import (
 	"github.com/artefactual-sdps/enduro/internal/storage/types"
 )
 
-type InternalLocationFactory func(config *LocationConfig) (Location, error)
-
-var DefaultInternalLocationFactory = func(config *LocationConfig) (Location, error) {
-	return NewInternalLocation(config)
-}
-
-type LocationFactory func(location *goastorage.Location) (Location, error)
-
-var DefaultLocationFactory = func(location *goastorage.Location) (Location, error) {
-	return NewLocation(location)
-}
-
-type Location interface {
+type Locationer interface {
 	UUID() uuid.UUID
-	Bucket() *blob.Bucket
-	Close() error
+	Bucket(ctx context.Context) (*blob.Bucket, error)
 }
 
-type locationImpl struct {
+type Location struct {
```

Comment:
You can't control how your objects are created if you export them, that can be a source of bugs if the setup is complex. If unexported, users need to rely on your constructors so you fully own how they're built.

Users should be able to access its methods if they use the interface instead of the concrete types. The constructor can return an interface type, that I think is very common if the implementation is unexported, something I always forget to do.

Reply:
Okay, thanks. It looks like I'm not accessing `Location` from outside the package, so making it an internal struct shouldn't be a problem.  I'll return an interface type too.